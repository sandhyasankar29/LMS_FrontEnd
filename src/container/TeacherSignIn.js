import { React, useState } from 'react';
import axios from 'axios';
import { useHistory } from "react-router-dom";
import Swal from 'sweetalert2'
import {
    Box,
    Button,
    Grommet,
    TextInput,
    Image,
    Text,
    Tabs,
    Tab
} from 'grommet';
import { Hide, View, Home } from 'grommet-icons';
import signImage from '../assets/teacher.svg'
import { grommet } from 'grommet/themes';
import './GlobalVariables'
const TeacherSignIn = () => {
    const urlSignIn = 'http://'+{global,ip}.ip+':3001/signinteacher';
    const urlGetOtp = 'http://'+{global,ip}.ip+':3001/getotpsignin';
    const urlVerifyOtp = 'http://'+{global,ip}.ip+':3001/verifysigninotp'
    const Toast = Swal.mixin({
        toast: true,
        position: 'top-end',
        showConfirmButton: false,
        timer: 3000,
        timerProgressBar: true,
        didOpen: (toast) => {
            toast.addEventListener('mouseenter', Swal.stopTimer)
            toast.addEventListener('mouseleave', Swal.resumeTimer)
        }
    })
    const history = useHistory();
    const [username, setUsername] = useState("");
    const [password, setPassword] = useState("")
    const [reveal, setReveal] = useState(false);
    const [get, setGet] = useState(true)
    const [otp, setOTP] = useState()
    const [roll, setRoll] = useState()
    const onSubmitSignIn = () => {
        if (username && password) {
            axios.post(urlSignIn, { tuserid: username, password: password })
                .then(res => {
                    if ((res.data.status).length > 0) {
                        localStorage.setItem("tuserid", res.data.tuserid)
                        localStorage.setItem("chairperson", res.data.chairperson)
                        localStorage.setItem('firstname', res.data.firstname)
                        localStorage.setItem('lastname', res.data.lastname)
                        history.replace('/teacherdashboard')
                    }
                }).catch(error => {
                    Swal.fire({
                        icon: 'error',
                        title: 'Oops...',
                        text: 'Check your Credentials',
                        footer: 'Forgot Password? Try logging in with OTP!</a>'
                    })
                })
        }
    }
    const onSendOTP = () => {
        if (roll) {
            axios.post(urlGetOtp, { userid: roll })
                .then(res => {
                    if ((res.data)) {
                        setGet(false)
                        localStorage.setItem("tuserid", roll)
                        localStorage.setItem("chairperson", res.data.chair)
                        localStorage.setItem('firstname', res.data.firstname)
                        localStorage.setItem('lastname', res.data.lastname)
                    }
                }).catch(error => {
                    Toast.fire({
                        icon: 'error',
                        title: 'Invalid ID'
                    })
                })
        }
    }

    const onSubmitOTP = () => {
        if (otp) {
            axios.post(urlVerifyOtp, { otp: otp })
                .then(res => {
                    if ((res.data)) {
                        history.replace('/teacherdashboard')
                    }
                }).catch(error => {
                    Toast.fire({
                        icon: 'error',
                        title: 'Wrong OTP'
                    })
                })
        }
    }
    document.body.style.overflow = "hidden"
    return (
        <Grommet full theme={grommet}>
            <Box pad="small" justify="start" align="start" >
                <Button primary icon={<Home />} id="HomeBtn" label="Go to Home" hoverIndicator="light-1" onClick={() => { history.replace('/') }} />
            </Box>
            <Box fill align="center" justify="center">
                <Box height="small" width="large" style={{
                    marginBottom: "3%",
                }}>
                    <Image src={signImage} fit="contain" />
                </Box>
                <Tabs>
                    <Tab title="Credentials">
                        <Box width="medium" pad="small">
                            <Box
                                width="medium"
                                direction="row"
                                align="center"
                                round="small"
                                border
                                style={{
                                    marginBottom: "10%",
                                }}>
                                <TextInput
                                    plain
                                    placeholder="Username"
                                    name="username" type="name"
                                    value={username}
                                    onChange={(event) => setUsername(event.target.value)}

                                />
                            </Box>

                            <Box
                                width="medium"
                                direction="row"
                                align="center"
                                round="small"
                                border
                            >
                                <TextInput
                                    plain
                                    placeholder="Password"
                                    name="password" type="password"
                                    // eslint-disable-next-line react/jsx-no-duplicate-props
                                    type={reveal ? 'text' : 'password'}
                                    value={password}
                                    onChange={(event) => setPassword(event.target.value)}
                                />
                                <Button
                                    icon={reveal ? <View size="medium" /> : <Hide size="medium" />}
                                    onClick={() => setReveal(!reveal)}
                                />
                            </Box>

                            <Box direction="row" justify="between" margin={{ top: 'medium' }}>
                                <Text id="newAcc" color="purple" onClick={() => { history.push('/teacherregister') }} >Not Signed Up?</Text>
                                <Button data-testid="button" active={true} onClick={onSubmitSignIn} id="submitBtn" type="submit" label="Log In" primary />
                            </Box>
                            <Box data-testid="button2" pad="medium" justify="center" align="center" gap="medium">
                                <Button hoverIndicator="light-1">
                                    <Text color="purple" onClick={() => { history.push('/forgotpassword') }} >Forgot Password</Text>
                                </Button>
                            </Box>
                        </Box>
                    </Tab>
                    <Tab title="OTP" id="otpLogin">
                        {get ?
                            <div>
                                <TextInput

                                    style={{ marginTop: "10%" }}
                                    placeholder="Enter Roll No"
                                    name="roll"
                                    // eslint-disable-next-line react/jsx-no-duplicate-props
                                    type={'text'}
                                    value={roll}
                                    onChange={(event) => setRoll(event.target.value)}
                                />
                                <Button style={{ marginTop: "7%" }} active={true} onClick={onSendOTP} id="submitBtn" type="submit" label="Get OTP" primary /></div> : <Box pad="small" width="medium">
                                <Box
                                    width="medium"
                                    direction="row"
                                    align="center"
                                    round="small"
                                    border
                                >
                                    <TextInput
                                        plain
                                        placeholder="Enter OTP"
                                        name="otp" type="password"
                                        // eslint-disable-next-line react/jsx-no-duplicate-props
                                        type={reveal ? 'text' : 'password'}
                                        value={otp}
                                        onChange={(event) => setOTP(event.target.value)}
                                    />
                                    <Button
                                        icon={reveal ? <View size="medium" /> : <Hide size="medium" />}
                                        onClick={() => setReveal(!reveal)}
                                    />
                                </Box>

                                <Box direction="row" justify="between" margin={{ top: 'medium' }}>
                                    <Button hoverIndicator="light-1">
                                        <Box data-testid="button3" pad="small" direction="row" align="center" gap="small">
                                            <Text color="purple" onClick={() => { setGet(true) }} >Cancel</Text>
                                        </Box>
                                    </Button>
                                    <Button data-testid="button" id="submitOTP" active={true} onClick={onSubmitOTP} type="submit" label="Submit OTP" primary />
                                </Box>
                            </Box>}
                    </Tab>
                </Tabs>
            </Box>
        </Grommet>
    );
};

export default (TeacherSignIn);