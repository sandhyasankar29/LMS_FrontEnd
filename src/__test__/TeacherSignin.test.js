import React from 'react';
import ReactDOM from 'react-dom';
import TeacherSignIn from '../container/TeacherSignIn';
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'

afterEach(cleanup)
it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<TeacherSignIn></TeacherSignIn>, div)
})

it("matches snapshot", () => {
    const tree = renderer.create(<TeacherSignIn></TeacherSignIn>).toJSON();
    expect(tree).toMatchSnapshot();
})
