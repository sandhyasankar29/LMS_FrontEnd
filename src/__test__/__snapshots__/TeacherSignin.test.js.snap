// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="StyledGrommet-sc-19lkkz7-0 bWDPZI"
>
  <div
    className="StyledBox-sc-13pk1d4-0 HvGCm"
  >
    <button
      className="StyledButton-sc-323bzc-0 bexXh"
      id="HomeBtn"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      type="button"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 iQEJEP"
      >
        <svg
          aria-label="Home"
          className="StyledIcon-ofa7kd-0 dcacLZ"
          viewBox="0 0 24 24"
        >
          <path
            d="M1,11 L12,2 L23,11 M15,23 L15,15 L15,15 L9,15 L9,23 M4,23 L4,9 M20,23 L20,9"
            fill="none"
            stroke="#000"
            strokeWidth="2"
          />
        </svg>
        <div
          className="StyledBox__StyledBoxGap-sc-13pk1d4-1 hjorfk"
        />
        Go to Home
      </div>
    </button>
  </div>
  <div
    className="StyledBox-sc-13pk1d4-0 bxevCE"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 bJtzEt"
      style={
        Object {
          "marginBottom": "3%",
        }
      }
    >
      <img
        className="StyledImage-ey4zx9-0 iBeQsm"
        src="teacher.svg"
      />
    </div>
    <div
      className="StyledBox-sc-13pk1d4-0 dYebPD StyledTabs-a4fwxl-2 jQLYpX"
      role="tablist"
    >
      <div
        className="StyledBox-sc-13pk1d4-0 hiXrYh StyledTabs__StyledTabsHeader-a4fwxl-0 gwFYfs"
      >
        <button
          aria-expanded={true}
          aria-selected={true}
          className="StyledButton-sc-323bzc-0 ebqXLG"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          role="tab"
          type="button"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 cLWgPR StyledTab-sc-1nnwnsb-0 kVEPVO"
          >
            <span
              className="StyledText-sc-1sadyjn-0 gWegOF"
            >
              Credentials
            </span>
          </div>
        </button>
        <button
          aria-expanded={false}
          aria-selected={false}
          className="StyledButton-sc-323bzc-0 ebqXLG"
          id="otpLogin"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onMouseOut={[Function]}
          onMouseOver={[Function]}
          role="tab"
          type="button"
        >
          <div
            className="StyledBox-sc-13pk1d4-0 gdixjB StyledTab-sc-1nnwnsb-0 kVEPVO"
          >
            <span
              className="StyledText-sc-1sadyjn-0 gImgGD"
            >
              OTP
            </span>
          </div>
        </button>
      </div>
      <div
        aria-label=" Tab Contents"
        className="StyledTabs__StyledTabPanel-a4fwxl-1 ixxRXT"
        role="tabpanel"
      />
    </div>
  </div>
</div>
`;
